core/ecschema-editing/src/test/Differencing/Conflicts.test.ts

```diff
@@ -858,21 +776,183 @@ describe("Difference Conflict Reporting", () => {
       };
 
       const differences = await runDifferences(sourceSchema, targetSchema);
-      expect(findDifferenceEntries(differences, "TestItem")).has.lengthOf(0);
-      expect(findConflictItem(differences, "TestItem")).deep.equals({
-        code: ConflictCode.ConflictingPropertyName,
-        schemaType: "EntityClass",
-        itemName: "TestItem",
-        path: "TestProperty",
-        source: "NavigationProperty",
-        target: "StructProperty",
-        description: "Target class already contains a property with a different type.",
-        difference: {
-          direction: "Backward",
-          name: "TestProperty",
-          relationshipName: "ConflictSchema.TestRelationship",
-          type: "NavigationProperty",
+      await expect(findConflictItem(differences, "TestItem")).to.be.eventually.fulfilled.then((conflict) => {
+        expect(conflict).to.have.a.property("code", ConflictCode.ConflictingPropertyName);
+        expect(conflict).to.have.a.property("schemaType", "EntityClass");
+        expect(conflict).to.have.a.property("itemName", "TestItem");
+        expect(conflict).to.have.a.property("path", "TestProperty");
+        expect(conflict).to.have.a.property("source", "ConflictSchema.TestRelationship");
+        expect(conflict).to.have.a.property("target", "ConflictSchema.TestStruct");
+        expect(conflict).to.have.a.property("description", "Target class already contains a property with a different type.");
+      });
+    });
+
+    it("should find a conflict between on different kind of quantities on primitive properties", async () => {
```

Comment:
as noted the issue is when the persistence unit differs ... also this test name doesn't read very well.